android/app/src/main/java/io/imagekit/imagekitdemo/UploadFileActivity.kt

```diff
@@ -0,0 +1,141 @@
+package io.imagekit.imagekitdemo
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.google.gson.Gson
+import com.imagekit.android.ImageKit
+import com.imagekit.android.ImageKitCallback
+import com.imagekit.android.entity.UploadError
+import com.imagekit.android.entity.UploadPolicy
+import com.imagekit.android.entity.UploadResponse
+import com.imagekit.android.preprocess.VideoUploadPreprocessor
+import io.imagekit.imagekitdemo.databinding.ActivityUploadFileBinding
+import kotlinx.coroutines.launch
+import java.io.*
+
+
+class UploadFileActivity : AppCompatActivity(), ImageKitCallback, View.OnClickListener {
+
+    private var uploadResultDialog: AlertDialog? = null
+    private var loadingDialog: AlertDialog? = null
+
+    private var file: File? = null
+
+    private var binding: ActivityUploadFileBinding? = null
+    private val viewModel: UploadAuthViewModel by viewModels()
+
+    override fun onClick(v: View?) {
+        val intent = Intent().setType("*/*").setAction(Intent.ACTION_GET_CONTENT)
+        startActivityForResult(Intent.createChooser(intent, "Select a file"), 24)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == 24 && resultCode == RESULT_OK) {
+            val selectedFile = data?.data
+            if (selectedFile != null){
+                file = FileUtils.getPath(this, selectedFile)?.let { File(it) }
+                uploadFile()
+            }
+        }
+    }
```

Comment:
Getting an error while trying to upload a pdf file 

```
2024-06-03 11:11:03.384 16305-16305 AndroidRuntime          io.imagekit.imagekitdemo             E  
FATAL EXCEPTION: main
Process: io.imagekit.imagekitdemo, PID: 16305
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=24, result=-1, data=Intent { dat=content://com.android.providers.media.documents/... flg=0x1 }} to activity {io.imagekit.imagekitdemo/io.imagekit.imagekitdemo.UploadFileActivity}: java.lang.NullPointerException
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5323)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.NullPointerException
	at io.imagekit.imagekitdemo.FileUtils.getDataColumn(FileUtils.kt:144)
	at io.imagekit.imagekitdemo.FileUtils.getPath(FileUtils.kt:240)
	at io.imagekit.imagekitdemo.UploadFileActivity.onActivityResult(UploadFileActivity.kt:43)
	at android.app.Activity.dispatchActivityResult(Activity.java:8628)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5316)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5362) 
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67) 
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
	at android.os.Handler.dispatchMessage(Handler.java:106) 
	at android.os.Looper.loopOnce(Looper.java:201) 
	at android.os.Looper.loop(Looper.java:288) 
	at android.app.ActivityThread.main(ActivityThread.java:7872) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
```

Reply:
Fixed. Please review